### PR TITLE
fix: use stable, precalculated BFS for tree allocation paths

### PR DESCRIPTION
**Pull request by dbjorge**
Resolves 3rd task in #6

Also fixed a few related pre-existing bugs I noticed while testing:

- Previously, it treated masteries as having bidirectional graph edges, which meant you could traverse across them in clusters that have multiple notables (eg, from Path of the Warrior -> life master -> Constitution). Updated it to treat masteries as using directional graph edges to disallow this.
- Previously, it treated all root nodes as "active", but this could result in picking a non-shortest path in some edge cases (see the new test case in `tree_graph_test.go` L67). Separated the "active" and "root" inputs to enable the algorithm to better handle this case.

To put together the test cases, I also implemented functionality similar to PoB 1's "developer mode" feature to show node IDs as part of their hover tooltips. I'll send that as a separate PR, but if you care to validate the test cases, it'll be much easier to do so using that PR.

### CODE REVIEW

File: frontend/src/lib/types/index.d.ts [hunk 1]
```diff
@@ -350,6 +350,21 @@ export declare namespace fwd {
     WriteTo(w?: unknown): [number, Error];
   }
 }
+export declare namespace mod {
```
Comment (dbjorge) [inline frontend/src/lib/types/index.d.ts:353]: These came in automatically when the typings got regenerated to pick up the updated calculate signature. I wasn't sure if you'd prefer for me to leave them in vs omit them from the PR, I'm fine with whichever you prefer.
